src/main/java/api/kirana/register/transactions/helpers/AppConfig.java

```diff
@@ -1,14 +0,0 @@
-package api.kirana.register.transactions.helpers;
-
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Configuration;
-import org.springframework.web.client.RestTemplate;
-
-@Configuration
-public class AppConfig {
-
-    @Bean
-    public RestTemplate restTemplate() {
-        return new RestTemplate();
```

Comment:
Use ConnectionFactory here so that we can control timeouts for this resttemplate, you can check RestTemplateBuilderFactory class from frameworks

 HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
                requestFactory.setConnectTimeout(this.connectTimeout);
                requestFactory.setConnectionRequestTimeout(this.connectionRequestTimeout);
                return new RestTemplate(requestFactory);